ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -177,4 +268,11 @@ private void disconnect() throws XAException {
         }
     }
 
+    private void assertIfRecoveryScanWasStarted() throws XAException {
+        if (delegate == null) {
+            LOGGER.warn("XAResource.recover has to be called first with XAResource.TMSTARTRSCAN argument");
+            throw new XAException(XAException.XAER_PROTO);
```

Comment:
+1, that could be useful here. It would constrain this part of the behaviour behind the SPI and allow the rest of the calls to be unguarded. Good catch.